src/Core/Framework/App/Lifecycle/Persister/PermissionPersister.php

```diff
@@ -14,18 +16,20 @@
 #[Package('framework')]
 class PermissionPersister
 {
-    public function __construct(private readonly Connection $connection)
-    {
+    public function __construct(
+        private readonly Connection $connection,
+        private readonly Privileges $privileges,
+    ) {
     }
 
     /**
      * @internal only for use by the app-system
      */
-    public function updatePrivileges(?Permissions $permissions, string $roleId): void
+    public function updatePrivileges(?Permissions $permissions, string $appId): void
     {
         $privileges = $permissions ? $permissions->asParsedPrivileges() : [];
 
-        $this->addPrivileges($privileges, $roleId);
+        $this->privileges->setPrivileges($appId, $privileges, Context::createDefaultContext());
```

Comment:
as this is internal, can't we get the context from the calling side?

Reply:
Yeah I refactored already in a further pr

Reply:
I moved it up in to this pr :)